src/PowerShellEditorServices.Host/CodeLens/PesterCodeLensProvider.cs

```diff
@@ -33,6 +40,38 @@ public PesterCodeLensProvider(EditorSession editorSession)
         {
             _editorSession = editorSession;
             _symbolProvider = new PesterDocumentSymbolProvider();
+
+            DeterminePesterVersion();
+        }
+
+        /// <summary>
+        /// Used to determine the value of <see cref="_pesterV4_6_0_OrHigherAvailable"/> as a background task.
+        /// </summary>
+        private void DeterminePesterVersion()
+        {
+            Task.Run(() =>
+            {
+                using (var powerShell = System.Management.Automation.PowerShell.Create())
+                {
+                    powerShell.AddCommand("Get-Module")
+                              .AddParameter("ListAvailable")
```

Comment:
The thing I'm not crazy about here is that someone may have 4.6.0 installed but an earlier version loaded.  If we go this route, perhaps we should double check to make sure an earlier version of Pester is not loaded into PSIC.  If so we could import-module with `-MinimumVersion 4.6` to ensure that version is loaded.  Just thinking out loud here so take this with a grain of salt.

Reply:
Yes, I thinking this as well, but calling `Get-Command Invoke-Pester` is much more expensive and takes more than 2 seconds on my (modern) machine, calling `Get-Module` takes only a quarter of a second the first time. We do the same type of checking for other modules such as PSSA and Plaster.

Reply:
> What I would definitely not do is execute Remove-Module or Import-Module because at this point the user has lost control over his/her shell

If we're executing Pester, the user must at some point cede control of their shell. That's the fundamental tricky thing about the PSIC -- we have to share with the user. The question is to what extent.

My concern is that the PSIC means a user can just remove/replace Pester at any time, invalidating an earlier one-off query.

One possibility is that we just extend what we do now:
- Do a once off check for Pester 4.6.0 installation (and may need to check what version is loaded, if any)
- If Pester 4.6.0 is not available, we do what we do now
- If the test name has no variables, we do what we do now
- Otherwise, we try executing with the new Pester syntax and if there's a ParameterBindingException on Invoke-Pester, we fall back to what we do now.

As an aside, I feel like that's pretty complicated (and could add significant overhead) for what's not a huge win as a feature. But if Pester 4.6.0 is well adopted, this might get easier. I certainly think the line number invocation is a more natural way to do things.

Reply:
@rjmholt 

> Otherwise, we try executing with the new Pester syntax and if there's a ParameterBindingException on Invoke-Pester, we fall back to what we do now.

Actually we could just do that, and then warn about required version.  That exception also has a `ParameterName` property we could use to verify that it's really a version problem.

Reply:
seeing failed attempts in the integrated console does not make for a good UX/impression, I would rather pre-pend some `if($IsPester460){}else{}` logic before doing that. Please my suggested alternative in the other comments below.